interpreter/interpreter.go

```diff
@@ -169,88 +168,42 @@ func (itrp *Interpreter) doOptionStatement(ctx context.Context, s *semantic.Opti
+
+		// Retrieve an option with the name from the scope.
+		// If it exists and is an option, then set the option
+		// as it is from the prelude.
+		if opt, ok := scope.Lookup(a.Identifier.Name); ok {
+			if opt, ok := opt.(*values.Option); ok {
+				opt.Value = init
+				return opt, nil
+			}
+		}
+
+		// Create a new option and set it within the current scope.
+		v := &values.Option{Value: init}
+		scope.Set(a.Identifier.Name, v)
+		return v, nil
 	case *semantic.MemberAssignment:
 		init, err := itrp.doExpression(ctx, a.Init, scope)
 		if err != nil {
 			return nil, err
 		}
-		pkgName := a.Member.Object.(*semantic.IdentifierExpression).Name
-		return itrp.setOption(scope, pkgName, a.Member.Property, init)
-	default:
-		return nil, errors.Newf(codes.Internal, "unsupported assignment %T", a)
-	}
-}
-
-// setOption applies the option to an existing option or creates a new option on the current package if it doesn't already exist.
-func (itrp *Interpreter) setOption(scope values.Scope, pkg, name string, v values.Value) (values.Value, error) {
-	v = values.Option{Value: v}
-	set, err := scope.SetOption(pkg, name, v)
-	if err != nil {
-		return nil, err
-	}
-	if !set {
-		// Option does not belong to any existing package.
-		// Set it as a normal value on the current scope.
-		scope.Set(name, v)
-	}
-	itrp.modifiedOptions = append(itrp.modifiedOptions, optionMutation{
-		Package: pkg,
-		Name:    name,
-		Value:   v,
-	})
-	return v, nil
-}
 
-type optionMutation struct {
-	Package, Name string
-	Value         values.Value
-}
-
-func (itrp *Interpreter) mutateFunctionScope(f function) (function, error) {
-	// copy the scope so we can safely mutate it
-	f.scope = f.scope.Copy()
-	copyPackages(f.scope)
-	mutatedPkg := false
-	for _, mut := range itrp.modifiedOptions {
-		// Check if the function is defined in the scope of package that was mutated
-		if f.pkg.Name() == mut.Package {
-			if !mutatedPkg {
-				f.pkg = f.pkg.Copy()
-			}
-			mutatedPkg = true
-			f.pkg.SetOption(mut.Name, mut.Value)
-			continue
-		}
-		// Apply the option to the scope
-		_, err := f.scope.SetOption(mut.Package, mut.Name, mut.Value)
+		obj, err := itrp.doExpression(ctx, a.Member.Object, scope)
 		if err != nil {
-			return f, err
+			return nil, err
 		}
-	}
-	if mutatedPkg {
-		// Reapply the package values to the scope.
-		f.scope = values.NewNestedScope(f.scope.Pop(), f.pkg)
-	}
-	return f, nil
-}
 
-// copyPackages creates a copy of the scope and any packages in scope
-func copyPackages(scope values.Scope) {
-	if scope == nil {
-		return
-	}
-	scope.LocalRange(func(k string, v values.Value) {
-		if p, ok := v.(*Package); ok {
-			scope.Set(k, p.Copy())
+		pkg, ok := obj.(values.Package)
+		if !ok {
+			return nil, errors.Newf(codes.Invalid, "%s: cannot set option %q on non-package value", a.Location(), a.Member.Property)
 		}
-	})
-	copyPackages(scope.Pop())
+		pkg.SetOption(a.Member.Property, init)
+		v, _ := pkg.Get(a.Member.Property)
```

Comment:
Since its not immediatley clear can you add a comment that the reason we do a Get right after a Set is so that we return the `values.Option` instead of just the `init` value?

Reply:
I changed this code so now there's a `SetOption` utility that returns the option. It's more obvious what it is doing so this awkward code doesn't have to exist.